### PR TITLE
Fixed Add woocommerce v3 lightbox support to the active theme.

### PR DESCRIPTION
**Pull request by juanlopez4691**
From version 3 of woocommerce plugin, lightbox support has to be
declared for the theme.

See:
https://woocommerce.wordpress.com/2017/02/28/adding-support-for-woocommerce-2-7s-new-gallery-feature-to-your-theme/

### CODE REVIEW

File: src/WooCommerce.php [hunk 1]
```diff
@@ -16,6 +16,11 @@ class WooCommerce {
    * @implements admin_init
    */
   public static function init() {
+    // Add WooCommerce v3.x Gallery Lightbox support
+    // @see https://woocommerce.wordpress.com/2017/02/28/adding-support-for-woocommerce-2-7s-new-gallery-feature-to-your-theme/
+    // @see https://github.com/woocommerce/woocommerce/wiki/Enabling-product-gallery-features-(zoom,-swipe,-lightbox)-in-3.0.0
+    add_theme_support('wc-product-gallery-lightbox');
```
Comment (sun) [inline src/WooCommerce.php]: According to https://woocommerce.wordpress.com/2017/02/28/adding-support-for-woocommerce-2-7s-new-gallery-feature-to-your-theme/ declaring this support will actually re-instantiate the default/built-in lightbox of WooCommerce.

But isn't the whole point of the gallerya plugin to replace the built-in lightbox?

**Comment by juanlopez4691** [comment]
Pushed a new fixup. Can you review, please?

File: assets/scripts/script.js [hunk 2]
```diff
@@ -53,7 +53,7 @@
           thumbnail: true,
```
Comment (sun) [inline assets/scripts/script.js]: ⬆️ Wondering whether we should upgrade the selector for the product gallery, too?  There is a more precise class now, which means that we also don't need to test for the wrapping context anymore.

**Comment by juanlopez4691** [comment]
Pushed a fixup. Review, please?

**Comment by juanlopez4691** [comment]
Don't know how, I managed to not commit t a file involved in the fixup. Now is corrected.

File: assets/scripts/script.js [hunk 3]
```diff
@@ -48,14 +48,12 @@
...
       }
-      if ($('body').hasClass('woocommerce') && $('.images').length > 0) {
-        $('.images').lightGallery({
-          thumbnail: true,
-          showThumbByDefault: false,
-          subHtmlSelectorRelative: true,
-          selector: '.zoom'
-        });
-      }
+      $('.woocommerce-product-gallery__image').lightGallery({
```
Comment (sun) [inline assets/scripts/script.js]: The wrapping container for the product gallery is `.woocommerce-product-gallery`

There is also `.woocommerce-product-gallery--with-images` and we should check in which cases that class is set – probably if there are any images in the gallery (as opposed to none) – perhaps the update announcement blog post explains it as well?

Comment (juanlopez4691) [reply assets/scripts/script.js]: `.woocommerce-product-gallery--with-images` is not in the post, but it is set (conditionally) only in one place:

https://github.com/woocommerce/woocommerce/blob/103f674d185c6aff3ed18c9762825675f61d3151/templates/single-product/product-image.php#L28.

It looks like a rather specific selector, as it clearly indicates 'a product gallery with images'. I will use it then.

Pushed a new fixup with this correction.

**Comment by juanlopez4691** [comment]
Pushed a new fixup. Review, please?_

File: assets/scripts/script.js [hunk 4]
```diff
@@ -40,22 +40,18 @@
...
+        thumbnail: true,
+        showThumbByDefault: false,
+        subHtmlSelectorRelative: true,
+        selector: '.gallerya__image > a'
+      });
+      $('.woocommerce-product-gallery').lightGallery({
+        thumbnail: true,
+        showThumbByDefault: false,
+        subHtmlSelectorRelative: true,
+        selector: '.attachment-shop_single'
```
Comment (sun) [inline assets/scripts/script.js]: `img[class*="attachment-"]`

Comment (juanlopez4691) [reply assets/scripts/script.js]: Applied your suggestion and just pushed a fixup.

**Pull request merged**